.github/workflows/ci.yaml

```diff
@@ -65,30 +65,19 @@ jobs:
       - run: echo "JAVA8_HOME=${JAVA_HOME}" >> $GITHUB_ENV
       - uses: actions/setup-java@v3
         with:
-          java-version: 11
+          java-version: 17
           distribution: temurin
-      - run: echo "JAVA11_HOME=${JAVA_HOME}" >> $GITHUB_ENV
-      - name: Compile with Java 11
+      - name: Compile with Java 17 and run tests with Java 8
         shell: bash
         run: |
           set -x
           export JAVA_HOME=$JAVA11_HOME
           export PATH=${JAVA_HOME}/bin:$PATH
-          # Why not compile? It's because the process needs to package jar so
-          # that gapic-generator-java module can use testlib modules of gax.
+          # Maven surefire plugin lets us to specify the JVM when running tests via
+          # the "jvm" system property.
           mvn verify --batch-mode --no-transfer-progress -Dcheckstyle.skip \
-              -DskipTests -Dfmt.skip
-      - name: Run test with Java 8 for all modules except gapic-generator-java
-        shell: bash
-        run: |
-          set -x
-          export JAVA_HOME=$JAVA8_HOME
-          export PATH=${JAVA_HOME}/bin:$PATH
-          # useIncrementalCompilation=false to avoid recompiling classes
-          # generated by Java 11 compiler above.
-          mvn -B -ntp test  --projects '!gapic-generator-java' \
-              -Dcheckstyle.skip -Dmaven.compiler.useIncrementalCompilation=false \
-              -Dfmt.skip
```

Comment:
Note: We used to run two Maven commands, one to compile and the other to run tests. However, the same thing can be done in surefire's jvm system property.